app/src/js/components/Granules/bulk-granule-delete.js

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { bulkGranuleDelete } from '../../actions';
+import _config from '../../config';
+import { updateSelectedQueryIds } from '../../utils/bulk';
+import DefaultModal from '../Modal/modal';
+import TextArea from '../TextAreaForm/text-area';
+
+const { kibanaRoot } = _config;
+
+const defaultQuery = {
+  index: '',
+  query: '',
+  ids: [],
+  forceRemoveFromCmr: false
+};
+
+const BulkDeleteModal = ({
+  asyncOpId,
+  className,
+  dispatch,
+  error,
+  handleSuccessConfirm,
+  inflight,
+  onCancel,
+  requestId,
+  showModal,
+  selected = [],
+  success
+}) => {
+  const [query, setQuery] = useState(JSON.stringify(defaultQuery, null, 2));
+  const [errorState, setErrorState] = useState();
+
+  // TODO: is there a better way to do this?
+  updateSelectedQueryIds(setQuery, query, selected);
```

Comment:
If this really sucks or is a really bad way to do this, feel free to say so. It's just implementing a "nice to have" feature that isn't a core requirement of this ticket (selected granules on the overview page are automatically set as `ids` in the JSON request payload)

Reply:
You should be able to just do something like this:

```
  const [query, setQuery] = useState(defaultQuery);

  const currentQuery = JSON.stringify({
    ...query,
    ids: selected
  }, null, 2);
```
And then pass `currentQuery` as the value for the `TextArea`
Because when `query` or `selected` changes, it'll cause the component to rerender and `currentQuery` will be updated